js/sykepengesoknad/validering/validerGraderteArbeidssporsmal.js

```diff
@@ -0,0 +1,91 @@
+import { fraInputdatoTilJSDato, getLedetekst } from '@navikt/digisyfo-npm';
+import { fjernIndexFraTag, formaterEnkeltverdi } from '../felleskomponenter/sporsmal/fieldUtils';
+import {
+    FERIE_NAR_V2, FERIE_PERMISJON_UTLAND,
+    FERIE_V2,
+    HVOR_MANGE_TIMER_PER_UKE,
+    HVOR_MYE_HAR_DU_JOBBET,
+    HVOR_MYE_PROSENT,
+    HVOR_MYE_PROSENT_VERDI,
+    HVOR_MYE_TIMER,
+    HVOR_MYE_TIMER_VERDI,
+    JOBBET_DU_GRADERT, PERMISJON_NAR_V2, PERMISJON_V2,
+} from '../enums/tagtyper';
+import { getStillingsprosent } from '../../sykepengesoknad-gammel-plattform/aktiviteter-i-sykmeldingsperioden/BeregnetArbeidsgrad';
+import { JA } from '../enums/svarEnums';
+
+const leggIndexPaTag = (tag, index) => {
+    return `${tag}_${index}`;
+};
+
+export const hentFerieOgPermisjonperioder = (values) => {
+    const harHattFerie = formaterEnkeltverdi(values[FERIE_V2]) === JA;
+    const harHattPermisjon = formaterEnkeltverdi(values[PERMISJON_V2]) === JA;
+    const ferieperioder = harHattFerie ? values[FERIE_NAR_V2] : [];
+    const permisjonperioder = harHattPermisjon ? values[PERMISJON_NAR_V2] : [];
+
+    return [
+        ...ferieperioder,
+        ...permisjonperioder,
+    ].map((periode) => {
+        return {
+            fom: fraInputdatoTilJSDato(periode.fom),
+            tom: fraInputdatoTilJSDato(periode.tom),
+        };
+    });
+};
+
+const validerGraderteArbeidssporsmal = (sporsmal, values, soknad) => {
+    const feilmeldinger = {};
+    const graderteArbeidssporsmal = sporsmal
+        .filter((_sporsmal) => {
+            return fjernIndexFraTag(_sporsmal.tag) === JOBBET_DU_GRADERT;
+        })
+        .filter((_sporsmal) => {
+            return formaterEnkeltverdi(values[_sporsmal.tag]) === _sporsmal.kriterieForVisningAvUndersporsmal;
+        })
+        .filter((_sporsmal) => {
+            const feriesporsmal = soknad.sporsmal.find((spm) => {
+                return spm.tag === FERIE_V2;
+            });
+            const feriePermUtlandsporsmal = soknad.sporsmal.find((spm) => {
+                return spm.tag === FERIE_PERMISJON_UTLAND;
+            });
+            const hovedsporsmalTags = soknad.sporsmal.map((s) => {
+                return s.tag;
+            });
+            const harSvartPaFeriesporsmal = feriesporsmal
```

Comment:
du kan returnere her direkte ;) eller ville du ha et navn for å beskrive returverdien?

Reply:
Ja, var faktisk akkurat det jeg tenkte, for å unngå å kommentere...